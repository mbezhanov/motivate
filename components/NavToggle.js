import React from "react";
import {Animated, StyleSheet, TouchableWithoutFeedback, View} from 'react-native';
import { Octicons } from '@expo/vector-icons';
import Colors from '../constants/Colors';
import Sizes from '../constants/Sizes';
import PropTypes from 'prop-types';

const navToggle = ({ onPress, rotate }) => (
  <TouchableWithoutFeedback onPress={onPress}>
    <View style={styles.button}>
      <Animated.View style={{ transform: [{ rotate }] }}>
        <Octicons name="plus" size={Sizes.M} color={Colors.HIGHLIGHT}  />
      </Animated.View>
    </View>
  </TouchableWithoutFeedback>
);

navToggle.propTypes = {
  onPress: PropTypes.func,
  rotate: PropTypes.object,
};

const styles = StyleSheet.create({
  button: {
    width: Sizes.XL * 2,
    height: Sizes.XL * 2,
    borderRadius: Sizes.XL,
    textAlign: 'center',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Colors.BACKGROUND,
  }
});

export default navToggle;
