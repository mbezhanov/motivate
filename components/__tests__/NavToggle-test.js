import React from 'react';
import renderer from 'react-test-renderer';
import snapshotDiff from 'snapshot-diff';
import NavToggle from '../NavToggle';

jest.mock('react-native/Libraries/Components/Touchable/TouchableWithoutFeedback', () => 'TouchableWithoutFeedback');

jest.mock('@expo/vector-icons', () => {
  const React = require('react');
  class Octicons extends React.Component {
    render() {
      return React.createElement('Octicons', this.props, this.props.children)
    }
  }
  return { Octicons };
});

describe('NavToggle', () => {
  let tree;

  beforeEach(() => {
    tree = renderer.create(<NavToggle />);
  });

  it('renders correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('can use a custom onPress handler', () => {
    const stub = () => '123';
    const a = tree.toJSON();
    const b = renderer.create(<NavToggle onPress={stub} />).toJSON();
    const diff = snapshotDiff(a, b);
    expect(diff).toMatchSnapshot();
  });

  it('can use an externally animated value to perform a rotation animation', () => {
    const stub = {
      _children: [],
      _parent: 0,
      _config: {
        inputRange: [0, 1],
        outputRange: ['0deg', '225deg'],
      },
      _interpolation: () => {},
    };
    const a = tree.toJSON();
    const b = renderer.create(<NavToggle rotate={stub} />).toJSON();
    const diff = snapshotDiff(a, b);
    expect(diff).toMatchSnapshot();
  });
});
