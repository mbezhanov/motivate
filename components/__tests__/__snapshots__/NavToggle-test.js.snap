// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavToggle can use a custom onPress handler 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,6 +1,8 @@
- <TouchableWithoutFeedback>
+ <TouchableWithoutFeedback
+   onPress={[Function stub]}
+ >
    <View
      style={
        Object {
          \\"alignItems\\": \\"center\\",
          \\"backgroundColor\\": \\"#2b2b2b\\","
`;

exports[`NavToggle can use an externally animated value to perform a rotation animation 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -15,11 +15,25 @@
      <View
        style={
          Object {
            \\"transform\\": Array [
              Object {
-               \\"rotate\\": undefined,
+               \\"rotate\\": Object {
+                 \\"_children\\": Array [],
+                 \\"_config\\": Object {
+                   \\"inputRange\\": Array [
+                     0,
+                     1,
+                   ],
+                   \\"outputRange\\": Array [
+                     \\"0deg\\",
+                     \\"225deg\\",
+                   ],
+                 },
+                 \\"_interpolation\\": [Function _interpolation],
+                 \\"_parent\\": 0,
+               },
              },
            ],
          }
        }
      >"
`;

exports[`NavToggle renders correctly 1`] = `
<TouchableWithoutFeedback>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2b2b2b",
        "borderRadius": 28,
        "height": 56,
        "justifyContent": "center",
        "textAlign": "center",
        "width": 56,
      }
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "rotate": undefined,
            },
          ],
        }
      }
    >
      <Octicons
        color="#fff"
        name="plus"
        size={16}
      />
    </View>
  </View>
</TouchableWithoutFeedback>
`;
